template/clojure/Dockerfile

```diff
@@ -1,68 +1,35 @@
-FROM openjdk:8u121-jdk-alpine as builder
-# [ARM64]: If you are targeting ARM64, you will run into an error because the
-# image above does not yet support that architecture. See this issue for more
-# details: https://github.com/tessellator/openfaas-clojure-template/issues/6. 
-# To work around this, comment out the FROM line above and uncomment the one 
-# below. 
-# You will also need to make changes in the other stages. Each section section 
-# that needs your attention will start with "[ARM64]".
-#FROM balenalib/generic-aarch64-alpine-openjdk:8-20210225 as builder
-
-ENV CLOJURE_VERSION="1.10.3.822"
-
-RUN apk --no-cache add curl bash \
-    && curl -O https://download.clojure.org/install/linux-install-${CLOJURE_VERSION}.sh \
-    && chmod +x linux-install-${CLOJURE_VERSION}.sh \
-    && ./linux-install-${CLOJURE_VERSION}.sh
+FROM clojure:openjdk-17-tools-deps as builder
 
 WORKDIR /home/app/entrypoint
-COPY ./entrypoint/deps.edn ./deps.edn
-COPY ./function/deps.edn ../function/deps.edn
-
-RUN clojure -A:uberjar -Stree
-
-# During the build step, the dev dependencies from the function are fetched. I
-# do not know why. This line will go and fetch those dependencies as an
-# independent step so that if only the function code changes there will be no
-# need to fetch the dev dependencies.
-RUN cd ../function && clojure -A:dev -Stree && cd ../entrypoint
 
-COPY . /home/app
-
-RUN clojure -X:uberjar :jar Entrypoint.jar \
-    && if [ -f ../function/manifest.mf ] ; then echo "applying manifest" && jar umf ../function/manifest.mf ./Entrypoint.jar ; fi
-
-## -----------------------------------------------------------------------------
-
-FROM alpine:latest as fwatchdog-fetcher
+COPY ./entrypoint/deps.edn ./deps.edn
+COPY ./function/deps.edn ./function/manifest* ../function/
 
-WORKDIR /home
+RUN clojure -P
+RUN clojure -A:build -P
 
-ENV WATCHDOG_VERSION="0.8.3"
+WORKDIR /home/app/function
 
-# [ARM64]: The fwatchdog executable is prebuilt for various architectures. If
-# you are targeting ARM64, you will need to change the ARCH environment variable
-# to pull the correct executable.
-ENV ARCH=amd64
-# ENV ARCH=arm64
+RUN clojure -P
+COPY ./function/src ./src
 
-RUN apk add curl \
-    && curl -sSL https://github.com/openfaas/of-watchdog/releases/download/${WATCHDOG_VERSION}/fwatchdog-${ARCH} > fwatchdog \
-    && chmod +x ./fwatchdog
+WORKDIR /home/app/entrypoint
+COPY ./entrypoint/src ./src
+COPY ./entrypoint/classes ./classes
+COPY ./entrypoint/build.clj ./build.clj
 
-## -----------------------------------------------------------------------------
+RUN clojure -T:build uber
+RUN if [ -f ../function/manifest.mf ] ; then echo "applying manifest" && jar umf ../function/manifest.mf ./bin/Entrypoint.jar ; fi
```

Comment:
I do not have time to check out this, but it might be possible to remove this possibility and supply the required values for the manifest as a map somewhere and merge them in the `build.clj` file

Reply:
👍 We can worry about this later.